### PR TITLE
4 ncnews

### PR DESCRIPTION
**Pull request by RyanJMarsh**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+describe("GET: /api/articles/:article_id/comments", () => {
+  test("200: should return comment for given article", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then(({ body }) => {
+        body.comments.forEach((comment) => {
+          expect(comment).toMatchObject({
+            comment_id: expect.any(Number),
+            author: expect.any(String),
```
Comment (johnhutcheon) [inline __tests__/app.test.js:103]: Could we be more specific with the author to expect the username of the author please?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+            article_id: 1,
+            votes: expect.any(Number),
+            body: expect.any(String),
+            created_at: expect.any(String)
+          });
+        });
+      });
+  });
```
Comment (johnhutcheon) [inline __tests__/app.test.js:111]: Please add one more 200 test for a valid ID which has no comments, so would return an empty array of comments 👍

File: controllers/app.controllers.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    .catch((error) => {
+      next({ status: 400, msg: "article_id is not of correct type" });
+    });
+};
+
+exports.getCommentsByArticleId = (request, response, next) => {
+  const { article_id } = request.params;
+  getCommentsByArticleIdData(article_id)
+    .then((comments) => {
+      if (comments.length === 0) {
```
Comment (johnhutcheon) [inline controllers/app.controllers.js:48]: This error handling is good, but please it move it into the model 👍

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Great work Ryan! Just check my comment about moving the error. It looks like this also needs to be done for Ticket 5 from Jodie's feedback, so please implement this. 
Let me know once you've done it with a message on Slack and I'll approve 👍

File: __tests__/app.test.js (hunk 1 above)
Comment (RyanJMarsh) [reply __tests__/app.test.js:103]: Not sure how to make it more specific? It's the comment author not article author so it won't all be the same for a particular article_id

Comment (johnhutcheon) [reply __tests__/app.test.js:103]: Sorry my bad, ignore this comment!
